### PR TITLE
Add soviet03

### PR DESCRIPTION
**Pull request by ghost**
This adds the third mission of the soviet campaign.

### CODE REVIEW

File: mods/ra/maps/soviet-03/soviet03.lua [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+		SpyHideout4.UnloadPassengers()
+		Trigger.AfterDelay(DateTime.Seconds(1), function()
+			TheSpy.Move(SpyGoal.Location)
+		end)
+	end
+end
+
+SpyHelicopterEscape = function()
+	if not spyHelicopterEscape then
+		spyHelicopterEscape  = true
```
Comment (clemty) [inline mods/ra/maps/soviet-03/soviet03.lua]: two spaces before `=` - better:
```spyHelicopterEscape = true```

File: mods/ra/maps/soviet-03/soviet03.lua [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+		UserInterface.SetMissionText("Time Remaining:" .. Utils.FormatTime(remainingTime), player.Color)
+		remainingTime = remainingTime - 1
+	elseif remainingTime == 0 and not spyReachedHideout4 then
+		UserInterface.SetMissionText("")
+		enemy.MarkCompletedObjective(alliedObjective)
+	elseif remainingTime == 0 and spyReachedHideout4 then
+		UserInterface.SetMissionText("")
+		SpyHelicopterEscape()
+	end
+end
```
Comment (clemty) [inline mods/ra/maps/soviet-03/soviet03.lua]: there should be another line break there

**Comment by Biofreak1987** [comment]
Updated

File: mods/ra/maps/soviet-03/soviet03.lua [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+			Media.PlaySpeechNotification(player, "ReinforcementsArrived")
+		end
+	end
+end)
+
+Trigger.OnEnteredFootprint(RTrapTrigger, function(a, id)
+	if not rTrapTrigger and a.Owner == player then
+		rTrapTrigger = true
+		Actor.Create("camera", true, { Owner = player, Location = CameraFinalArea.Location })
+		RSoldier3.Attack(RSoldierTrap)
```
Comment (abcdefg30) [inline mods/ra/maps/soviet-03/soviet03.lua]: I think we need an `IsDead` check here.
```
if not RSoldierTrap.IsDead then
	RSoldier3.Attack(RSoldierTrap)
	RSoldier4.Attack(RSoldierTrap)
end

File: mods/ra/maps/soviet-03/soviet03.lua [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+		Media.PlaySpeechNotification(player, "WarningFourMinutesRemaining")
+	elseif remainingTime == DateTime.Minutes(3) then
+		Media.PlaySpeechNotification(player, "WarningThreeMinutesRemaining")
+	elseif remainingTime == DateTime.Minutes(2) then
+		Media.PlaySpeechNotification(player, "WarningTwoMinutesRemaining")
+	elseif remainingTime == DateTime.Minutes(1) then
+		Media.PlaySpeechNotification(player, "WarningOneMinuteRemaining")
+	end
+	if remainingTime > 0 and timerstarted then
+		UserInterface.SetMissionText("Time Remaining:" .. Utils.FormatTime(remainingTime), player.Color)
```
Comment (abcdefg30) [inline mods/ra/maps/soviet-03/soviet03.lua]: Please add a space after the `:`.

**Comment by abcdefg30** [comment]
I'd also suggest killing the barrels at the base entry in `IntroSequence`.

**Comment by Biofreak1987** [comment]
Updated, added a few more IsDead checks than suggested... like for the spy

**Comment by clemty** [comment]
afaik mission Lua files should also be listed in the Microsoft Visual Studio Solution File (OpenRA.sln)

**Comment by Biofreak1987** [comment]
Rebased and added an entry to OpenRA.sln

File: mods/ra/maps/soviet-03/soviet03.lua [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+		if not TheSpy.IsDead and not SpyHideout2.IsDead then
+			TheSpy.EnterTransport(SpyHideout2)
+		end
+	end
+end)
+
+Trigger.OnEnteredFootprint(SpyHideout2Trigger, function(a, id)
+	if not spyHideout2Trigger and a.Owner == player then
+		spyHideout2Trigger = true
+		SpyGuards1 = Reinforcements.Reinforce(germany, EnemyReinforcements1SpyHideout2, { EnemyReinforcements1.Location, EnemyReinforcements1Goal.Location }, 0)
```
Comment (abcdefg30) [inline mods/ra/maps/soviet-03/soviet03.lua]: We either need a check here if the house in which those guys are hiding is still there, or you expand the Footprint, as you can snipe the house with grenadiers before triggering the cell trigger.

**Comment by abcdefg30** [comment]
Other than that. :+1: / :white_check_mark:

**Comment by Biofreak1987** [comment]
Moved the Footprint further away from the house

**Comment by Mailaender** [comment]
Works as promised. :white_check_mark: Well done!

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/525b2474979096ced5084df4a6abbba0fbbb5b1b...35b0ab8a411eeec383ea43c687f531eda5c4c4d1)

**Pull request merged**